### PR TITLE
Vega support

### PR DESCRIPTION
**Pull request by mcmcgrath13**
This PR proposes a few changes to increase support for Vega specifications:
 * Update the MIME type for VGSpecs to "application/vnd.vega.v4+json", which is the current standard for JupyterLab (have to install an extension for the current v3 to show interactively)
* Create an `AbstractVegaSpec` type to simplify some mostly duplicate functions
* Add a section to the docs on using Vega and how to use it in a function (I had some trouble getting this to work and it feels slightly hack-y - @fredo-dedup  do you have a better way to do this?)

I noticed the examples folder seems to use an outdated syntax - can that be removed as well?

This PR shouldn't be merged until the corresponding [IJulia PR](https://github.com/JuliaLang/IJulia.jl/pull/821) is merged so they are ready to support the v4 mime type.

I expect this will need to be updated shortly as Vega recently released v5.

### CODE REVIEW

**Comment by mcmcgrath13** [comment]
Travis is failing on mac due to an issue with the Cairo build - I need to dig into this

File: Project.toml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name = "VegaLite"
```
Comment (davidanthoff) [inline Project.toml]: I'd prefer not to have a `Project.toml` here for now. What I have seen happening is that the CI tests use this one here, and then folks forget to update the `REQUIRE` file, and then one can actually end up with releasing a version and it passes all CI that doesn't work on end user systems. Once the whole attobot infrastructure is moved over to `Project.toml`, we can obviously add this, so would be great if we can save this somehow!

**Comment by davidanthoff** [comment]
> Travis is failing on mac due to an issue with the Cairo build - I need to dig into this

That unfortunately happens regularly... It is really frustrating, the whole BinDeps.jl binary provider story that Cairo.jl utilizes is just not reliable and seems to break _a lot_...  I think the only proper solution is that someone creates a BinaryBuilder.jl style solution for Cairo.jl and Rsvg.jl. But that implies Gtk.jl also needs to be ported, and at that point it is a mega project...

In any case, that failure is clearly not related to your PR, so I think we can merge things here independently of the Cairo.jl situation.

**Comment by mcmcgrath13** [comment]
@davidanthoff that's good to know about Cairo... (kind of - it's always nicer when things just work).  I've gitignored the project.toml and when this is merged I'll do another PR with just the Project.toml and the updated travis script so it can be merged whenever attobot is ready.

On the IJulia side, I've updated that PR to preemptively include the mime types for vega v5 and vegalite v3 so we should be able to switch those over whenever they're supported by jupyter.

**Comment by davidanthoff** [comment]
Super, thanks a lot!

**Pull request merged**